frontend/src/metabase/common/components/DashboardPicker/components/DashboardPicker.tsx

```diff
@@ -0,0 +1,222 @@
+import type { Ref } from "react";
+import { useCallback, useState, forwardRef, useImperativeHandle } from "react";
+import { useDeepCompareEffect } from "react-use";
+
+import {
+  skipToken,
+  useGetCollectionQuery,
+  useGetDashboardQuery,
+} from "metabase/api";
+import { isValidCollectionId } from "metabase/collections/utils";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { useSelector } from "metabase/lib/redux";
+import { getUserPersonalCollectionId } from "metabase/selectors/user";
+import type {
+  ListCollectionItemsRequest,
+  CollectionItemModel,
+  Dashboard,
+} from "metabase-types/api";
+
+import { CollectionItemPickerResolver } from "../../CollectionPicker/components/CollectionItemPickerResolver";
+import { getPathLevelForItem } from "../../CollectionPicker/utils";
+import {
+  LoadingSpinner,
+  NestedItemPicker,
+  type PickerState,
+} from "../../EntityPicker";
+import type { DashboardPickerOptions, DashboardPickerItem } from "../types";
+import { getCollectionIdPath, getStateFromIdPath, isFolder } from "../utils";
+
+export const defaultOptions: DashboardPickerOptions = {
+  showPersonalCollections: true,
+  showRootCollection: true,
+  allowCreateNew: true,
+};
+interface DashboardPickerProps {
+  onItemSelect: (item: DashboardPickerItem) => void;
+  initialValue?: Pick<DashboardPickerItem, "model" | "id">;
+  options: DashboardPickerOptions;
+  models?: CollectionItemModel[];
+  shouldDisableItem?: (item: DashboardPickerItem) => boolean;
+}
+
+const useGetInitialCollection = (
+  initialValue?: Pick<DashboardPickerItem, "model" | "id">,
+) => {
+  const isDashboard =
+    initialValue && ["dashboard"].includes(initialValue.model);
+
+  const dashboardId = isDashboard ? Number(initialValue.id) : undefined;
+
+  const { data: currentDashboard, error: dashboardError } =
+    useGetDashboardQuery(
+      dashboardId
+        ? {
+            id: dashboardId,
+          }
+        : skipToken,
+    );
+
+  const collectionId =
+    isDashboard && currentDashboard
+      ? currentDashboard?.collection_id
+      : initialValue?.id;
+
+  const { data: currentCollection, error: collectionError } =
+    useGetCollectionQuery(
+      !isDashboard || !!currentDashboard
+        ? (isValidCollectionId(collectionId) && collectionId) || "root"
```

Comment:
this line feels like it should just be it's own variable so we don't have the nested pseudo ternary thing